src/shared/components/Button/Button.style.ts

```diff
@@ -72,53 +75,116 @@ const colorsFromProps = ({ variant }: ButtonStyleProps) => {
   switch (size) {
-    case 'smaller': {
-      padding = '10px'
+    case 'small': {
+      padding = '9.5px 12px'
       fontSize = typography.sizes.button.small
+      buttonSizeWithoutText = '32px'
       break
     }
-    case 'small': {
-      padding = hasText ? `12px 14px` : '12px'
+    case 'medium': {
+      padding = hasText ? `12px 16px` : '12px'
       fontSize = typography.sizes.button.medium
+      buttonSizeWithoutText = '40px'
       break
     }
-    case 'regular':
+    case 'large':
     default: {
-      padding = hasText ? `14px 20px` : '14px'
+      padding = hasText ? `15px 20px` : '14px'
       fontSize = typography.sizes.button.large
+      buttonSizeWithoutText = '48px'
       break
     }
   }
   return css`
-    width: ${full ? '100%' : ''};
+    width: ${!hasText ? buttonSizeWithoutText : full ? '100%' : ''};
+    height: ${!hasText ? buttonSizeWithoutText : 'auto'};
     display: ${full ? 'flex' : 'inline-flex'};
     font-size: ${fontSize};
     padding: ${padding};
   `
 }
 
-const disabled = ({ disabled }: ButtonStyleProps) =>
-  disabled
+const disabled = ({ disabled, variant }: ButtonStyleProps) => {
+  let bgColor, color, borderColor, opacity
+  switch (variant) {
+    case 'tertiary': {
+      bgColor = colors.black
+      color = colors.white
+      borderColor = colors.transparent
+      opacity = 0.32
+      break
+    }
+    case 'secondary': {
+      bgColor = colors.black
+      color = colors.white
+      borderColor = colors.white
+      opacity = 0.32
+      break
+    }
+    case 'primary':
+    default: {
+      bgColor = colors.blue[500]
+      color = colors.white
+      borderColor = colors.blue[500]
+      opacity = 0.4
+      break
+    }
+  }
+  return disabled
     ? css`
         box-shadow: none;
-        color: ${colors.gray[200]};
-        background-color: ${colors.gray[400]};
-        border-color: ${colors.gray[400]};
+        color: ${color};
+        background-color: ${bgColor};
+        border-color: ${borderColor};
+        opacity: ${opacity};
         &:hover {
-          color: ${colors.gray[200]};
-          background-color: ${colors.gray[400]};
-          border-color: ${colors.gray[400]};
+          color: ${color};
+          background-color: ${bgColor};
+          border-color: ${borderColor};
+          opacity: ${opacity};
         }
         &:active {
-          color: ${colors.gray[200]};
-          background-color: ${colors.gray[400]};
-          border-color: ${colors.gray[400]};
+          color: ${color};
+          background-color: ${bgColor};
+          border-color: ${borderColor};
+          opacity: ${opacity};
         }
       `
     : null
+}
+const iconSizeFromProps = ({ size: sizeProp = 'medium', hasText }: IconStyleProps) => {
+  let size, margin
+  if (!hasText) {
+    size = 'auto'
+  } else {
+    switch (sizeProp) {
+      case 'small': {
+        size = typography.sizes.icon.small
+        margin = '-9.5px 0'
```

Comment:
We should put those values in some variables since they're used in more than one place. This way it will be easy to miss this when changing the paddings